static/js/drafts.js

```diff
@@ -128,12 +128,20 @@ export function confirm_delete_all_drafts() {
     });
 }
 
-export function rename_topic(stream_id, old_topic, new_topic) {
+export function rename_stream_recipient(old_stream_id, new_stream_id, old_topic, new_topic) {
     const current_drafts = draft_model.get();
     for (const draft_id of Object.keys(current_drafts)) {
         const draft = current_drafts[draft_id];
-        if (util.same_stream_and_topic(draft, {stream_id, topic: old_topic})) {
-            draft.topic = new_topic;
+        if (util.same_stream_and_topic(draft, {stream_id: old_stream_id, topic: old_topic})) {
+            // If new_stream_id is undefined, that means the stream wasn't updated.
+            if (new_stream_id !== undefined) {
+                draft.stream_id = new_stream_id;
+                draft.stream = sub_store.get(new_stream_id).name;
```

Comment:
it would be great to change this so that drafts in localstorage don't store both stream_id and stream, and instead calculate the name from the id -- I almost left this without updating the name, which seemed to work okay in the app for the cases I was testing but would have been at the very least a technical bug

Reply:
I suspect the issue is that it's possible just have `anno` in the stream input box since it's a text field, and we don't want to lose what you typed there when saving a draft.

Once we complete our UI plan was to change the stream input field to a `dropdown_list_widget`, it would eliminate the possibility of invalid partial input there, at which point we could have the drafts system ignore the legacy `stream` field (and then it'll age away naturally with the autodeletion).

Reply:
May be worth your adding a commit with some TODO comments about this detail; up to you.

Reply:
ah that makes sense! another great reason to move to dropdown :)